packages/grafana-data/src/text/text.ts

```diff
@@ -24,7 +24,7 @@
 }
 
 const cleanNeedle = (needle: string): string => {
-  return needle.replace(/[[{(][\w,.-?:*+]+$/, '');
+  return needle.replace(/[[{(][\w,.\/0-9:;<=>?:*+]+$/, '');
```

Comment:
looking at the tests, the purpose of this regexp it to somehow clean another regexp string? but it's only replacing a single opening brace followed by alphanum plus a bunch of other chars?

would this not leave the resulting regexp broken since it's not removing any trailing braces? is there some sane explanation of what kinds of needles this is meant to guard against?

Reply:
i don't think any of the tests in [1] cover what `cleanNeedle()` is expected to clean?

[1] https://github.com/grafana/grafana/blob/1fa0fb5d64d46f0af8fee20236727bbc3ce6542f/packages/grafana-data/src/text/text.test.ts

Reply:
also, `0-9` is already covered by `\w`